--- gsasl-1.8.0/examples/saml20/Makefile.am.orig	2012-05-21 08:22:18.000000000 +0200
+++ gsasl-1.8.0/examples/saml20/Makefile.am	2014-05-10 09:34:34.745200455 +0200
@@ -22,7 +22,7 @@
 	$(LASSO_CFLAGS)
 AM_LDFLAGS = -no-install
 LDADD = $(top_builddir)/lib/src/libgsasl.la $(top_builddir)/gl/libgl.la \
-	$(LASSO_LIBS)
+	$(LASSO_LIBS) -lgobject-2.0
 
 EXTRA_DIST = README
 EXTRA_DIST += gsasl-saml20-sp.php
